smac/facade/smac_facade.py

```diff
@@ -340,16 +340,71 @@ def optimize(self):
         return incumbent
 
     def get_tae_runner(self):
+        '''
+            returns target algorithm evaluator (TAE) object
+            which can run the target algorithm given a
+            configuration
+
+            Returns
+            -------
+            smac.tae.execute_ta_run.ExecuteTARun
+        '''
         return self.solver.intensifier.tae_runner
 
     def get_runhistory(self):
+        '''
+            returns the runhistory 
+            (i.e., all evaluated configurations and the results)
+
+            Returns
+            -------
+            smac.runhistory.runhistory.RunHistory
+        '''
         if not hasattr(self, 'runhistory'):
             raise ValueError('SMAC was not fitted yet. Call optimize() prior '
                              'to accessing the runhistory.')
         return self.runhistory
 
     def get_trajectory(self):
+        '''
+            returns the trajectory 
+            (i.e., all incumbent configurations over time)
+
+            Returns
+            -------
+            List of entries with the following fields: 
+            'train_perf', 'incumbent_id', 'incumbent',
+            'ta_runs', 'ta_time_used', 'wallclock_time'
+        '''
+
         if not hasattr(self, 'trajectory'):
             raise ValueError('SMAC was not fitted yet. Call optimize() prior '
                              'to accessing the runhistory.')
         return self.trajectory
+
+    def get_X_y(self):
```

Comment:
This code looks a lot like a copy from `RunHistory2EPM`. Maybe that code can be reused?